src/v/cluster/cloud_data/aggregated_uploader.cc

```diff
@@ -0,0 +1,571 @@
+                co_await wait_for_next_upload();
+                /*TODO: remove*/ vlog(
+                  _logger.info,
+                  "[schedule_uploads] "
+                  "upload_data_flow_controller.schedule_uploads acquiring "
+                  "units");
+                auto u = co_await ss::get_units(_mutex, 1);
+                /*TODO: remove*/ vlog(
+                  _logger.info,
+                  "[schedule_uploads] "
+                  "upload_data_flow_controller.schedule_uploads acquiring "
+                  "units done");
+                auto size_bytes = _bytes_used;
+                _bytes_used = 0;
+                decltype(_pending) to_upload;
+                std::swap(_pending, to_upload);
+                u.return_all();
+                // TODO: fixme
+                // This code is performing simple round robin
+                // search to schedule uploads from different shards.
+                auto shard = ss::shard_id(shard_rr % ss::smp::count);
+                shard_rr++;
+                auto result = ss::make_lw_shared<aggregated_write_request>();
+                result->to_upload = std::move(to_upload);
+                result->size_bytes = size_bytes;
+                /*TODO: remove*/ vlog(
+                  _logger.info,
+                  "[schedule_uploads] "
+                  "upload_data_flow_controller.schedule_uploads ~~~ target: "
+                  "{}, "
+                  "upload size: {}",
+                  shard,
+                  result->size_bytes);
+                if (size_bytes > 0) {
+                    co_await _parent->container().invoke_on(
+                      shard,
+                      [fp = ss::make_foreign(std::move(result)),
+                       req = std::move(to_upload)](
+                        aggregated_uploader<Clock>& svc) mutable {
+                          /*TODO: remove*/ vlog(
+                            clusterlog.info,
+                            "NEEDLE [schedule_uploads] calling 'upload_object' "
+                            "on "
+                            "shard {}",
+                            ss::this_shard_id());
+                          return svc.upload_object(std::move(fp));
+                      });
+                }
+            }
+        } catch (...) {
+            /*TODO: remove*/ vlog(
+              _logger.info,
+              "[schedule_uploads] upload_data_flow_controller.schedule_uploads "
+              "failure {}",
+              std::current_exception());
+        }
+        /*TODO: remove*/ vlog(
+          _logger.info,
+          "[schedule_uploads]upload_data_flow_controller.schedule_uploads "
+          "exit");
+        co_return;
+    }
+    ss::future<> collect_data() {
+        /*TODO: remove*/ vlog(
+          _logger.info,
+          "[collect_data] upload_data_flow_controller.collect_data");
+        auto h = _gate.hold();
+        try {
```

Comment:
This method together with collect_data seems to be exception-safety sensitive. This try-catch-all block make me nervous. Shall we simplify it by moving cirtical things into a noexcept method and wrapping only things we are comfortable with failing in the try-catch-all block?